tools/manifest/tests/test_manifest.py

```diff
@@ -254,6 +254,31 @@ def test_reftest_computation_chain_update_test_type():
     assert list(m) == [("testharness", test2.path, {test2})]
 
 
+def test_reftest_computation_chain_update_node_change():
+    m = manifest.Manifest()
+
+    s1 = SourceFileWithTest("test1", "0"*40, item.RefTest, [("/test2", "==")])
+    s2 = SourceFileWithTest("test2", "0"*40, item.RefTestNode, [("/test3", "==")])
+
+    assert m.update([s1, s2]) is True
+
+    test1 = s1.manifest_items()[1][0]
+    test2 = s2.manifest_items()[1][0]
+
+    assert list(m) == [("reftest", test1.path, {test1}),
+                       ("reftest_node", test2.path, {test2})]
+
+    #test2 changes to support type
+    s3 = SourceFileWithTest("test2", "1"*40, item.SupportFile)
```

Comment:
So I think you want to redefine `s2` here rather than creating a new `s3`

Reply:
I made the changes and it works as it should now!
Still need to understand why, though. 
Thanks a lot :smile: 

And I'm sorry I made a new PR for this, I'll ensure I push to the same branch from now on.